packages/core-modules/test/contracts/modules/AssociatedSystemsModule.test.js

```diff
@@ -0,0 +1,239 @@
+const { ethers } = hre;
+const assert = require('assert/strict');
+const assertRevert = require('@synthetixio/core-js/utils/assertions/assert-revert');
+const { bootstrap } = require('@synthetixio/deployer/utils/tests');
+const initializer = require('@synthetixio/core-modules/test/helpers/initializer');
+
+const toBytes32 = ethers.utils.formatBytes32String;
+
+describe('AssociatedSystemsModule', () => {
+  const { proxyAddress } = bootstrap(initializer, {
+    modules: '.*(Owner|Upgrade|AssociatedSystems).*',
+  });
+
+  let AssociatedSystemsModule;
+  let owner, user;
+
+  before('identify signers', async () => {
+    [owner, user] = await ethers.getSigners();
+  });
+
+  before('identify modules', async () => {
+    AssociatedSystemsModule = await ethers.getContractAt('AssociatedSystemsModule', proxyAddress());
+  });
+
+  describe('registerUnmanagedSystem()', async () => {
+    it('only callable by owner', async () => {
+      await assertRevert(
+        AssociatedSystemsModule.connect(user).registerUnmanagedSystem(
+          toBytes32('wohoo'),
+          owner.address
+        ),
+        'Unauthorized'
+      );
+    });
+
+    it('registers unmanaged', async () => {
+      await AssociatedSystemsModule.connect(owner).registerUnmanagedSystem(
```

Comment:
add tests to verify the events are emited correctly

Reply:
https://github.com/Synthetixio/synthetix-v3/pull/1043/commits/3d1ab3d6afc720cf3de153c16727d2ce16356f7c